controllers/userController.ts

```diff
@@ -0,0 +1,60 @@
+//\\//\\//\\//\\//\\//\\//\\//\\//\\//\\//\\
+// USER ENTITY UPDATE & DELETE FUNCTIONS  \\
+//\\//\\//\\//\\//\\//\\//\\//\\//\\//\\//\\
+
+import { RequestHandler, Request, Response, NextFunction } from "express";
+import bcrypt from "bcrypt";
+
+// Import TS types
+import { UserMethods } from "../models/userModel";
+import { User } from "@prisma/client";
+
+// Extend express-session SessionData to include user data
+declare module "express-session" {
+    interface SessionData {
+        user: User;
+        loggedIn: boolean;
+    }
+}
+
+// Update controller
+// -------------------------------------------------------------------------
+
+export const update =
+    (
+        getUserByEmail: UserMethods.GetUserByEmail,
+        getUserByUserName: UserMethods.GetUserByUserName,
+        updateUser: UserMethods.UpdateUser
+    ): RequestHandler =>
+    async (
+        req: Request,
+        res: Response,
+        next: NextFunction
+    ): Promise<void | Response> => {
+        try {
+            // Check if username and email are already taken
+            const userNameExists = await getUserByUserName(req?.body?.userName);
+            const emailExists = await getUserByEmail(req?.body?.email);
+            if (
+                (userNameExists &&
+                    req?.session?.user?.id !== userNameExists?.id) ||
+                (emailExists && req?.session?.user?.id !== emailExists?.id)
+            )
+                return res.status(400).json({
+                    status: "error",
+                    message:
+                        "A user already ``   exists with that username or email address",
+                });
+
+            // Encrypt the password
+            const password = req.body.password;
+            const hashedPassword = bcrypt.hashSync(password, 6);
+            req.body.password = hashedPassword;
+            const result = await updateUser(req?.session?.user?.id, req.body);
+            console.log("Updated user " + req.body.userName + " successfully!");
+            return res.status(201).json({ status: "success", data: result });
+        } catch (error) {
+            console.log(`userController.ts ERROR: ${error}`);
```

Comment:
logging error messages can be useful in development, but it clutters the console in tests and production. I would suggest either remove the log and only add it in debugging, or wrap it in an if statement to only run in dev mode --> `if(process.env.NODE_ENV === 'development') ...

Reply:
All of these points have been implemented in the current commit, thanks for your attention to detail.